internal/providers/azure/azure.go

```diff
@@ -67,7 +67,7 @@ func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
 
 	logger.Debug("mounting config device")
 	if err := logger.LogOp(
-		func() error { return unix.Mount(devicePath, mnt, "udf", unix.MS_RDONLY, "") },
+		func() error { return unix.Mount(devicePath, mnt, "udf,iso9660", unix.MS_RDONLY, "") },
```

Comment:
I have a feeling that this comma-list is some sugar specific to the `mount` utility, but not something that the syscall directly likes. My reading of the docs may be wrong though.

Reply:
Yeah, indeed you're right @lucab. I threw this up as an early draft. During my deep dive on WALinuxAgent, Cloud-init I suspected that the volume might be an `iso9660`. I had briefly looked at the API docs `unix.Mount` and I wondered it would support this syntax. 

An strace of `mount -t udf,is09660...` shows this:

```
stat("/sbin/mount.udf", 0x7ffc675bd420) = -1 ENOENT (No such file or directory)
stat("/sbin/fs.d/mount.udf", 0x7ffc675bd420) = -1 ENOENT (No such file or directory)
stat("/sbin/fs/mount.udf", 0x7ffc675bd420) = -1 ENOENT (No such file or directory)
mount("/dev/loop0", "/var/mnt", "udf", MS_SILENT, NULL) = -1 EINVAL (Invalid argument)
stat("/sbin/mount.iso9660", 0x7ffc675bd420) = -1 ENOENT (No such file or directory)
stat("/sbin/fs.d/mount.iso9660", 0x7ffc675bd420) = -1 ENOENT (No such file or directory)
stat("/sbin/fs/mount.iso9660", 0x7ffc675bd420) = -1 ENOENT (No such file or directory)
mount("/dev/loop0", "/var/mnt", "iso9660", MS_SILENT, NULL) = -1 EACCES (Permission denied)
stat("/sbin/mount.udf", 0x7ffc675bd420) = -1 ENOENT (No such file or directory)
stat("/sbin/fs.d/mount.udf", 0x7ffc675bd420) = -1 ENOENT (No such file or directory)
stat("/sbin/fs/mount.udf", 0x7ffc675bd420) = -1 ENOENT (No such file or directory)
mount("/dev/loop0", "/var/mnt", "udf", MS_RDONLY|MS_SILENT, NULL) = -1 EINVAL (Invalid argument)
stat("/sbin/mount.iso9660", 0x7ffc675bd420) = -1 ENOENT (No such file or directory)
stat("/sbin/fs.d/mount.iso9660", 0x7ffc675bd420) = -1 ENOENT (No such file or directory)
stat("/sbin/fs/mount.iso9660", 0x7ffc675bd420) = -1 ENOENT (No such file or directory)
mount("/dev/loop0", "/var/mnt", "iso9660", MS_RDONLY|MS_SILENT, NULL) = 0
openat(AT_FDCWD, "/usr/share/locale/locale.alias", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=2998, ...}) = 0
read(3, "# Locale name alias data base.\n#"..., 4096) = 2998
```
And with that, its clear this is mount sugar.